controller/gen/apis/serviceprofile/v1alpha2/types.go

```diff
@@ -28,9 +28,10 @@ type ServiceProfile struct {
 
 // ServiceProfileSpec specifies a ServiceProfile resource.
 type ServiceProfileSpec struct {
-	Routes       []*RouteSpec   `json:"routes"`
-	RetryBudget  *RetryBudget   `json:"retryBudget,omitempty"`
-	DstOverrides []*WeightedDst `json:"dstOverrides,omitempty"`
+	Routes             []*RouteSpec   `json:"routes"`
+	RetryBudget        *RetryBudget   `json:"retryBudget,omitempty"`
+	DstOverrides       []*WeightedDst `json:"dstOverrides,omitempty"`
+	FullyQualifiedName string         `json:"fullyQualifiedName,omitempty"`
```

Comment:
isn't the name of the service profile already the fully qualified name?  can we use that instead of creating a new field for it?

Reply:
@adleong Yes if a Service Profile exists for the service, then the name would already be the FQN. But, if a Service Profile does not exist for the service then we want to be able to return a default profile and we wouldn't have a name for it.

Adding this field ensures that by default, a service with or without a Service Profile always has the FQN field populated.

Reply:
ah, I see that we're populating this in the traffic split adapter which seems like a really unexpected place to handle this.  

seems like we should take the FQN from the name of the service profile if it exists.  if it doesn't exist, we'll call `defaultServiceProfile()`.  If the profile translator holds the service ID (just like the traffic split adaptor does) then we can populate the FQN on the default service profile from the service ID.